src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java

```diff
@@ -190,12 +190,8 @@ private String proxyPost(Request request, Response response) {
                 *
                 * Other requests will still be proxied, just not stored.
                 */
-                QueryVariables queryVariables = getPOJOFromJSON(requestBody, Query.class).variables;
-
-                // Follows the method used in otp-ui core-utils storage.js
-                String randomBatchId = Integer.toString((int) (Math.random() * 1_000_000_000), 36);
-
-                if (!handlePlanTripResponse(randomBatchId, queryVariables, otpDispatcherResponse, otpUser)) {
+                Query query = getPOJOFromJSON(requestBody, Query.class);
+                if (!handlePlanTripResponse(query.batchId, query.variables, otpDispatcherResponse, otpUser)) {
```

Comment:
I'm assuming `handlePlanTripResponse` is able to handle batch id missing?

Reply:
That is the case. With this PR,`batchId` is an optional field when parsing already-stored trip requests, and `handlePlanTripResponse` already sets `batchId` to "Batch Id not provided by caller." if not provided.